alacritty_terminal/src/term/mod.rs

```diff
@@ -124,6 +138,16 @@ impl<'a> RenderableSearch<'a> {
     }
 }
 
+/// Common rendering primitives stored in one structure to simplify passing them around.
+struct RenderableContentInner<'a, T, C> {
+    term: &'a Term<T>,
+    selection: Option<SelectionRange<Line>>,
+    search: RenderableSearch<'a>,
+    config: &'a Config<C>,
+    cursor_point: Point,
+    cursor_shape: CursorShape,
```

Comment:
We don't pass cursor here, since when getting rendering cursor we need to get cell first before getting color. I also decided to not create like 2 different renderable cursor and decided to stick with one.